src/main/java/io/jenkins/plugins/conventionalcommits/utils/ProjectTypeFactory.java

```diff
@@ -0,0 +1,30 @@
+package io.jenkins.plugins.conventionalcommits.utils;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class ProjectTypeFactory {
+
+    static Map<String, ProjectType> projectTypeMap = new HashMap<>();
```

Comment:
You might not need to do this; you can just instantiate the class when you identify which one it should be.